module-2/backend/week_one.md

```diff
@@ -1,34 +1,64 @@
+Get - reading information
+Post - Create information
+Put - Replace information
+Patch - Modify information
+Delete - Delete information
 2. What is Sinatra?
-4. What is MVC?
-5. Why do we follow conventions when creating our actions/path names in our Sinatra routes?
-6. What types of variables are accessible in our view templates without explicitly passing them?
-7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
-  
+Sintra is a web-applicaiton framework for ruby.
+3. What is MVC?
+MVC is a specific model for designing web-applications. It stands for Models, views, controller.
+4. Why do we follow conventions when creating our actions/path names in our Sinatra routes?
+We follow conventions for the same reason that we follow conventions in software development; we need our code to be easily understood by other developers. Additionally, in the case of Sinatra routes, it is a way that all websites can standardize the common practices of creating, reading, updating and deleting information.
+5. What types of variables are accessible in our view templates without explicitly passing them?
+Instance variables
+6. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
+
   ```ruby
   get '/horses' do
-    erb :index
+    @count = 1
+    name = "Mr. Ed"
+    erb :index :locals => name: name
   end
   ```
 
-8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
+7. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
 9. What's the purpose of ERB?
+The purpose of Embedded Ruby is to insert ruby into a text document, usually into an HTML page.
 10. Why do I need a development AND test database?
+You need to have two databases so that you can separate your your development and testing suite. For testing you would probably want to use fixtures to cut down on the time that your tests take. In development you might want to use a fuller database to closely mirror what your application will be using.
 11. What's responsive design?
+Responsive design is opposed to fixed design. In my mind, a fixed design would be the same across all different platforms and clients. Whereas a responsive design changes dynamically to fit the needs of each individual client/platform. An example would be a screen that would resize itself across smartphones, desktops etc.
 12. What is CRUD and why is it important?
-13. What does HTTP stand for? 
+CRUD stands for Create, Read, Update, and Delete. This is important because these describes the way in which a user can interact with a given database.
+13. What does HTTP stand for?
+  Hypertext Transfer Protocol
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?

```

Comment:
<%= ruby %> and <% ruby %> 

First way allows ruby data to show on a web page and second does not